### PR TITLE
Preparatory patches to start using the module interface for all components

### PR DESCRIPTION
**Pull request by ranj063**

### CODE REVIEW

File: src/audio/module_adapter/module_adapter.c [hunk 1]
```diff
@@ -733,3 +733,87 @@ void module_adapter_free(struct comp_dev *dev)
 	rfree(mod);
 	rfree(dev);
 }
+
```
Comment (dbaluta) [inline src/audio/module_adapter/module_adapter.c:757]: We could use here a better explanation about why do we need set_large_config and get_..

Comment (ranj063) [reply src/audio/module_adapter/module_adapter.c:757]: fixed.

File: src/audio/module_adapter/module_adapter.c [hunk 2]
```diff
@@ -35,7 +35,21 @@ struct comp_dev *module_adapter_new(const struct comp_driver *drv,
...
+	case SOF_COMP_MODULE_ADAPTER:
+	{
+		struct ipc_config_process *ipc_module_adapter = spec;
+
+		size = ipc_module_adapter->size;
+		data = ipc_module_adapter->data;
+		break;
+	}
+	default:
+		return NULL;
```
Comment (lyakh) [inline src/audio/module_adapter/module_adapter.c:51]: do we expect to add more types here? I was in favour of using `switch` for all enumerations even if only one case has to be handled, but we do use `if` for such cases too and I must admit that looks simpler, shorter and just as readable, so maybe here
```
if (config->type != SOF_COMP_MODULE_ADAPTER)
	return NULL;
```
would be easier too, but that's totally up to you

**Review by lyakh — CHANGES_REQUESTED** [review]

File: src/audio/module_adapter/module_adapter.c (hunk 2 above)
Comment (paulstelian97) [reply src/audio/module_adapter/module_adapter.c:51]: I believe there _is_ indeed such an expectation so I'd say this is fine.

Comment (ranj063) [reply src/audio/module_adapter/module_adapter.c:51]: yes @lyakh the adapter will need to handle other types. The switch/case will be expanded as and when I convert the current components

**Review by lgirdwood — APPROVED** [review]

**Review by lyakh — APPROVED** [review]

File: src/audio/module_adapter/module_adapter.c [hunk 3]
```diff
@@ -35,7 +35,21 @@ struct comp_dev *module_adapter_new(const struct comp_driver *drv,
...
+	uint32_t size;
+
+	switch (config->type) {
+	case SOF_COMP_MODULE_ADAPTER:
+	{
+		struct ipc_config_process *ipc_module_adapter = spec;
+
+		size = ipc_module_adapter->size;
+		data = ipc_module_adapter->data;
+		break;
```
Comment (dbaluta) [inline src/audio/module_adapter/module_adapter.c:48]: @ranj063 there is a problem with this code. For codec_adapter we never used config->type. Does the codec_adapter tests still work for Intel usecases?

I'm confused about who sets config->type from the Linux kernel side.


Cc: @cujomalainey

Comment (ranj063) [reply src/audio/module_adapter/module_adapter.c:48]: @dbaluta all components have their comp type set in the topology except the module_adapter comps. So we need to set that in topology and parse it in the kernel with ABI compatibility checks

Comment (dbaluta) [reply src/audio/module_adapter/module_adapter.c:48]: @ranj063 thanks for the explanation!

**Pull request merged**